src/deepsparse/yolo/schemas.py

```diff
@@ -67,10 +69,53 @@ def from_files(cls, files: List[str], **kwargs) -> "YOLOInput":
     class Config:
         arbitrary_types_allowed = True
 
+    @staticmethod
+    def create_test_inputs(
+        batch_size: int = 1,
+    ) -> Dict[str, Union[Union[List[str], str, List[Any]]]]:
+        """
+        Create and return a test input for this schema
+
+        :param batch_size: The batch_size of inputs to return
+        :return: A dict representing inputs for Yolo pipeline
+        """
+
+        sample_image_path = Path(__file__).parents[0] / "sample_images" / "basilica.jpg"
+        sample_image_abs_path = str(sample_image_path.absolute())
+
+        images = [sample_image_abs_path for _ in range(batch_size)]
+        return {"images": images}
+
+    def split(self) -> Generator["YOLOInput", None, None]:
+        """
+        Split a current `YOLOInput` object with a batch size b, into a
+        generator of b smaller objects with batch size 1, the returned
+        object can be iterated on.
+
+        :return: A Generator of smaller `YOLOInput` objects each
+            representing an input of batch-size 1
+        """
 
-class YOLOOutput(BaseModel):
+        images = self.images
+
+        # case 1: do nothing if single input of batch_size 1
+        if isinstance(images, str):
+            yield self
+
+        elif isinstance(images, list) and len(images) and isinstance(images[0], str):
+            # case 2: List[str, Any] -> multiple images of size 1
+            for image in images:
+                yield YOLOInput(
+                    images=image,
+                )
+
+        else:
+            raise ValueError(f"Could not breakdown {self} into smaller batches")
```

Comment:
why do we need to raise here?

Reply:
The YOLOInput Schema also accepts a List[Any], it is ambiguous how to split the input then.

Reply:
sounds fine for now, let's keep an eye out and revisit if this condition is hit in practice